[Merged by Bors] - chore(Logic): reduce use of autoImplicit

grunweg opened the pull request:
In one case, replacing this naively errored, so I just merged the single declaration using it.

Delete two unused variables in Logic/Basic.


---

Commits can be reviewed individually.

<!-- The text above the `---` will become the commit message when your
PR is merged. Please leave a blank newline before the `---`, otherwise
GitHub will format the text above it as a title.

To indicate co-authors, include lines at the bottom of the commit message
(that is, before the `---`) using the following format:

Co-authored-by: Author Name <author@email.com>

Any other comments you want to keep out of the PR commit should go
below the `---`, and placed outside this HTML comment, or else they
will be invisible to reviewers.

If this PR depends on other PRs, please list them below this comment,
using the following format:
- [ ] depends on: #abc [optional extra text]
- [ ] depends on: #xyz [optional extra text]
-->

[![Open in Gitpod](https://gitpod.io/button/open-in-gitpod.svg)](https://gitpod.io/from-referrer/)

semorrison commented on Mathlib/Logic/Equiv/Defs.lean:
```diff
@@ -320,7 +317,7 @@ theorem Perm.coe_subsingleton {α : Type*} [Subsingleton α] (e : Perm α) : (e
 #align equiv.apply_eq_iff_eq Equiv.apply_eq_iff_eq
 
-theorem apply_eq_iff_eq_symm_apply (f : α ≃ β) : f x = y ↔ x = f.symm y := by
+theorem apply_eq_iff_eq_symm_apply {x : α} {y : β}(f : α ≃ β) : f x = y ↔ x = f.symm y := by
```
```suggestion
theorem apply_eq_iff_eq_symm_apply {x : α} {y : β} (f : α ≃ β) : f x = y ↔ x = f.symm y := by
```

semorrison commented on Mathlib/Logic/Function/Basic.lean:
```diff
@@ -807,12 +806,15 @@ end Extend
 
-theorem comp_left {f : α → β} {g : α → γ} (h : FactorsThrough g f) (g' : γ → δ) :
+set_option autoImplicit true in
+theorem comp_left {f : α → β} {g : α → γ}  (h : FactorsThrough g f) (g' : γ → δ) :
```
```suggestion
theorem comp_left {f : α → β} {g : α → γ} (h : FactorsThrough g f) (g' : γ → δ) :
```

semorrison commented:
I'm a bit skeptical of replacing `set_option autoImplicit` with `set_option ... in ...`, but I guess it is an improvement.

bors d+

grunweg commented:
@semorrison Thanks for the review; I have fixed your nits.
You raise a fair point: hence, I have gone ahead and also removed the `set_option autoImplicit true`s this PR introduces.
I'll let it sit for a day in case you want to take a look again, and merge if there are no comments.

grunweg commented:
Since I have not further comments, let me
bors r+
